openquake/risklib/connectivity.py

```diff
@@ -53,77 +122,609 @@ def analysis(dstore):
+            ignore_index=True)
+
+        # To store the sum of performance indicator at nodal level to calulate
+        # the average afterwards
+        ccl_table['CCL_node'] = 1 - ccl_table['CNS']
+        ccl_table1 = ccl_table.drop(columns=['CNO', 'CNS'])
+        ccl_table1 = ccl_table1.reset_index()
+        dem_cl = pd.concat((dem_cl, ccl_table1)).groupby(
+            'id', as_index=False).sum()
+
+        pcl_table1 = pcl_table.drop(columns=['NS0', 'NS'])
+        pcl_table1 = pcl_table1.reset_index()
+        dem_cl = pd.concat((dem_cl, pcl_table1)).groupby(
+            'id', as_index=False).sum()
+
+        wcl_table1 = wcl_table.drop(columns=['WS0', 'WS'])
+        wcl_table1 = wcl_table1.reset_index()
+        dem_cl = pd.concat((dem_cl, wcl_table1)).groupby(
+            'id', as_index=False).sum()
+
+        eff_table1 = eff_table.drop(columns=['Eff0', 'Eff'])
+        eff_table1 = eff_table1.reset_index()
+        node_el = pd.concat((node_el, eff_table1)).groupby(
+            'id', as_index=False).sum()
+
+    return (dem_cl, node_el, event_connectivity_loss_ccl,
+            event_connectivity_loss_pcl, event_connectivity_loss_wcl,
+            event_connectivity_loss_eff)
+
+
+def EFLWCLPCLloss_TAZ(exposure_df, G_original, TAZ_nodes,
+                      eff_nodes, damage_df, g_type):
+    # When the nodes acts as both demand and supply.
+    # For example, traffic analysis zone in transportation network. This
+    # calculates, efficiency loss (EFL),
+    # weighted connectivity loss (WCL),partial connectivity loss(PCL).
+    # Simple connectivity loss (SCL) doesnt make any sense in this case.
+
+    # To store the information of the performance indicators at connectivity
+    # level
+    taz_cl = exposure_df[exposure_df['supply_or_demand'] == 'TAZ'].iloc[:, 0:1]
+    node_el = exposure_df[exposure_df['type'] == 'node'].iloc[:, 0:1]
+
+    pcl_table = pd.DataFrame({'id': TAZ_nodes})
+    wcl_table = pd.DataFrame({'id': TAZ_nodes})
+    eff_table = pd.DataFrame({'id': eff_nodes})
+    eff_table.set_index("id", inplace=True)
+    pcl_table.set_index('id', inplace=True)
+    wcl_table.set_index('id', inplace=True)
+
+    # Create an empty dataframe with columns "event_id" and
+    # "CCL"/"PCL"/"WCL"/"EFFLoss"
+    event_connectivity_loss_pcl = pd.DataFrame(columns=['event_id', 'PCL'])
+    event_connectivity_loss_wcl = pd.DataFrame(columns=['event_id', 'WCL'])
+    event_connectivity_loss_eff = pd.DataFrame(columns=['event_id', 'EFFLoss'])
+
+    # To check the the values for each node before the earthquake event
+
+    # For calculating partial connectivity loss
+    for i in TAZ_nodes:
+        count = 0
+        for j in TAZ_nodes:
+            if i != j:
+                if nx.has_path(G_original, j, i):
+                    count = count + 1
+        pcl_table.at[i, 'NS0'] = count
+
+    # Code below is not giving correct answer because the path is checked from
```

Comment:
Can we remove these lines #583–588?

Reply:
yes we can remove these comments. Now from line 598 to 603

Reply:
Done